engines/sherlock/inventory.cpp

```diff
@@ -0,0 +1,524 @@
+		const Common::String &description, const Common::String &examine) :
+		_requiredFlag(requiredFlag), _name(name), _description(description),
+		_examine(examine), _lookFlag(0) {
+}
+
+/**
+ * Synchronize the data for an inventory item
+ */
+void InventoryItem::synchronize(Common::Serializer &s) {
+	s.syncAsSint16LE(_requiredFlag);
+	s.syncAsSint16LE(_lookFlag);
+	s.syncString(_name);
+	s.syncString(_description);
+	s.syncString(_examine);
+}
+
+/*----------------------------------------------------------------*/
+
+Inventory::Inventory(SherlockEngine *vm) : Common::Array<InventoryItem>(), _vm(vm) {
+	Common::fill(&_invShapes[0], &_invShapes[MAX_VISIBLE_INVENTORY], (ImageFile *)nullptr);
+	_invGraphicsLoaded = false;
+	_invIndex = 0;
+	_holdings = 0;
+	_invMode = INVMODE_EXIT;
+}
+
+Inventory::~Inventory() {
+	freeGraphics();
+}
+
+/**
+ * Free inventory data
+ */
+void Inventory::freeInv() {
+	freeGraphics();
+
+	_names.clear();
+	_invGraphicsLoaded = false;
+}
+
+/**
+ * Free any loaded inventory graphics
+ */
+void Inventory::freeGraphics() {
+	for (uint idx = 0; idx < MAX_VISIBLE_INVENTORY; ++idx)
+		delete _invShapes[idx];
+
+	Common::fill(&_invShapes[0], &_invShapes[MAX_VISIBLE_INVENTORY], (ImageFile *)nullptr);
+	_invGraphicsLoaded = false;
+}
+
+/**
+ * Load the list of names the inventory items correspond to, if not already loaded, 
+ * and then calls loadGraphics to load the associated graphics
+ */
+void Inventory::loadInv() {
+	// Exit if the inventory names are already loaded
+	if (_names.size() > 0)
+		return;
+
+	// Load the inventory names
+	Common::SeekableReadStream *stream = _vm->_res->load("invent.txt");
+
+	int streamSize = stream->size();
+	while (stream->pos() < streamSize) {
+		Common::String name;
+		char c;
+		while ((c = stream->readByte()) != 0)
+			name += c;
+
+		_names.push_back(name);
+	}
+
+	delete stream;
+
+	loadGraphics();
+}
+
+/**
+ * Load the list of names of graphics for the inventory
+ */
+void Inventory::loadGraphics() {
+	if (_invGraphicsLoaded)
+		return;
+
+	// Default all inventory slots to empty
+	Common::fill(&_invShapes[0], &_invShapes[MAX_VISIBLE_INVENTORY], (ImageFile *)nullptr);
+
+	for (int idx = _invIndex; (idx < _holdings) && (idx - _invIndex) < MAX_VISIBLE_INVENTORY; ++idx) {
+		// Get the name of the item to be displayed, figure out its accompanying
+		// .VGS file with its picture, and then load it
+		int invNum = findInv((*this)[idx]._name);
+		Common::String fName = Common::String::format("item%02d.vgs", invNum + 1);
+
+		_invShapes[idx - _invIndex] = new ImageFile(fName);
+	}
+
+	_invGraphicsLoaded = true;
+}
+
+/**
+ * Searches through the list of names that correspond to the inventory items
+ * and returns the numer that matches the passed name
+ */
+int Inventory::findInv(const Common::String &name) {
+	for (int idx = 0; idx < (int)_names.size(); ++idx) {
+		if (name.equalsIgnoreCase(_names[idx]))
+			return idx;
+	}
+
+	return 1;
```

Comment:
Is this really supposed to return `1` and not `-1` or something to indiciate not found?